pointcloud/basic_octree.go

```diff
@@ -35,7 +37,7 @@ type BasicOctree struct {
 type basicOctreeNode struct {
 	nodeType NodeType
 	children []*BasicOctree
-	point    PointAndData
+	point    *PointAndData
```

Comment:
changing to pointer to allow nil checks of empty data

Reply:
Indeed